database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java

```diff
@@ -243,6 +228,16 @@ protected String getErrorDetailsProviderClassName() {
     return null;
   }
 
+  /**
+   * Returns the DBErrorDetailsProvider instance.
+   * Override this method to provide a custom DBErrorDetailsProvider instance.
+   *
+   * @return DBErrorDetailsProvider instance
+   */
+  protected DBErrorDetailsProvider getErrorDetailsProvider() {
+    return new DBErrorDetailsProvider();
```

Comment:
why do we want to return a new instance on every call?

Can't we just create it once if null, otherwise reuse the same object everywhere?